projects/canopy/src/lib/forms/sort-code/sort-code.component.ts

```diff
@@ -114,6 +109,14 @@ export class LgSortCodeComponent implements OnInit, ControlValueAccessor {
   }
 
   ngOnInit(): void {
+    /* this is needed to sync validation when the parent form is first submitted
+       without having any values in the sort code inputs */
+    this.invalidSortCode$?.pipe(take(1))?.subscribe(() => {
```

Comment:
Is there maybe a way of setting the formGroup as required whenever the three fields are empty?

Reply:
It is set to required and it is marked as invalid, it's just that the sort code input fields are still seen as valid when the check from `IsControlInvalid` runs. I suspect it has something to do with the fact that the form has the option to `updateOn: 'blur'` so I suspect if no value is present there, it doesn't update properly as it's not being funnelled through the `blur` logic.  I also checked to see if we have the option to maybe put two values for the `updateOn` key but it only accepts one of these three 😢 `'blur', 'submit', 'onChange'` and in our case we need to have `blur` and `submit`.